Properly display degrees in mathML(MatrixExpression)

d-torrance opened the pull request:
@mikestillman - this fixes the matrix display bug we were seeing in Jupyter in texmacs mode.

### Before
![image](https://github.com/user-attachments/assets/d83f9d4d-ce8e-4399-9eeb-e07ad64003c4)

### After
![image](https://github.com/user-attachments/assets/7a496584-b13d-4a79-9610-36cb023432a0)

We also replace the *'s with ·'s when displaying products:
![image](https://github.com/user-attachments/assets/29768b1f-ecf7-40de-89d4-ed6787f61f5c)

mikestillman commented:
@d-torrance Thanks for fixing this!  

In displaying polynomials, I was wondering if perhaps we would prefer to not print an asterisk or a dot?  What do you think?

Sometimes there is no dot, sometimes there is.  I think it has to do with whether there is an exponent (so something like x * y^2, but x^2 y).  I will look at the output more carefully to see what seems preferable.

mikestillman commented:
I can either accept this PR as is, and we can work on the spacing later, or, if you have ideas, we can make those changes now.

mikestillman commented on M2/Macaulay2/m2/mathml.m2:
```diff
@@ -59,7 +59,29 @@ mathML FunctionApplication := m -> (
-mathML MatrixExpression := x -> concatenate( "<mrow><mo>(</mo>", mtableML x, "<mo>)</mo></mrow>", newline )
+
+-- zero-width "phantom" element to make sure that degrees line up
+mphantom := x -> concatenate(
```
We don't need to do this now, but fixing the spacing between terms (currently the + or - spacing is less than inside a monomial.  Also, sometimes the dot (or previously, asterisk) is included, sometimes not).  See conversation for an example...

d-torrance replied:
I think the big problem with how we represent these `Product` objects (not just by `mathML`, but in `net(Product)` too) is not knowing what the factors are.  We could totally represent `Product{x, y}` by `xy`, but we want `Product{2,3}` to be `2 * 3` and not `23`.

mikestillman reviewed (commented):
I can accept this as is.  We can play with the spacing mentioned in the review either for this PR or later.

d-torrance commented:
@mikestillman -- I've updated it so we only use &middot; in `factor(ZZ)`, i.e., immediately after a `Power` object with an exponent of 1:

![image](https://github.com/user-attachments/assets/b20836ec-145e-4516-9cfb-2f661d5cf092)

d-torrance commented:
I changed my mind and switched `mathML(Product)` so that it's essentially identical to `texMath(Product)` for consistency:

![image](https://github.com/user-attachments/assets/febe0314-4aa5-44a5-96b4-21c3d99ba241)

mikestillman commented:
Thanks, this latest version displays polynomials in texmacs mode in jupyter much better!

mikestillman reviewed (approved):
This looks good to me, thanks!
